fcrepo-kernel-impl/src/main/java/org/fcrepo/kernel/impl/rdf/impl/ReferencesRdfContext.java

```diff
@@ -48,21 +57,36 @@ public ReferencesRdfContext(final FedoraResource resource,
         throws RepositoryException {
         super(resource, idTranslator);
         property2triple = new PropertyToTriple(resource.getNode().getSession(), idTranslator);
-        concat(putStrongReferencePropertiesIntoContext());
-        concat(putWeakReferencePropertiesIntoContext());
+        session = resource.getNode().getSession();
+        putReferencesIntoContext(resource.getNode().getWeakReferences());
+        putReferencesIntoContext(resource.getNode().getReferences());
     }
 
-    private Iterator<Triple> putWeakReferencePropertiesIntoContext() throws RepositoryException {
-        final Iterator<Property> properties = resource().getNode().getWeakReferences();
+    private void putReferencesIntoContext(final Iterator<Property> properties) throws RepositoryException {
+        while (properties.hasNext()) {
```

Comment:
Yes, you want to ensconce your actual computation in an abstraction that can remain on the heap-- a function. Are you saying that there are times when your computation, given a `Property`, may not return a triple? May just say, "Nothing to do here-- next `Property`, please!"